### PR TITLE
Fix tabs across linewrap

### PR DESCRIPTION
**Pull request by chrisduerr**
This resolves an issue with tabs not breaking across line boundaries,
instead the characters would just all get written to the last column and
thus be lost.

It also tweaks the behavior of what happens when the terminal resizes
with the default tabspaces changed, using something like the `tabs`
program. Previously all tabstops would be reset to the default on
resize, which is what URxvt does. Now the tabspaces are kept and the new
columns are filled with the default tabstops, which emulates Termite.

### CODE REVIEW

File: alacritty_terminal/src/term/mod.rs [hunk 1]
```diff
@@ -1542,9 +1538,14 @@ impl<T: EventListener> Handler for Term<T> {
...
     }
 
+    /// Insert tab at cursor position.
     #[inline]
     fn put_tab(&mut self, mut count: i64) {
-        trace!("Putting tab: {}", count);
+        // A tab after the last column is the same as a linebreak
+        if self.input_needs_wrap {
+            self.wrapline();
+            return;
```
Comment (kchibisov) [inline alacritty_terminal/src/term/mod.rs]: I wasn't able to reach this code, could you share an example on how to do this?

Comment (chrisduerr) [reply alacritty_terminal/src/term/mod.rs]: `echo "\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt\tt"`

Comment (chrisduerr) [reply alacritty_terminal/src/term/mod.rs]: This is about tabs breaking across linewraps, so of course you need to make sure there are tabs breaking across lines.

Comment (kchibisov) [reply alacritty_terminal/src/term/mod.rs]: Oh, I see. I was doing similar thing but it wasn't working for me when I tried it first time for some reason.

**Pull request merged**